lib/computeStatistics.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+var Cesium = require('cesium');
+var isDataUri = require('./isDataUri');
+var PrimitiveHelpers = require('./PrimitiveHelpers');
+
+var defined = Cesium.defined;
+
+module.exports = {
+    getStatistics : getStatistics,
+    printStatistics : printStatistics
+};
+
+function getBuffersSize(buffers) {
+    var size = 0;
+    for (var id in buffers) {
+        if (buffers.hasOwnProperty(id)) {
+            size += buffers[id].byteLength;
+        }
+    }
+    return size;
+}
+
+function getNumberOfExternalProperties(property) {
+    var id;
+    var uri;
+    var count = 0;
+    for (id in property) {
+        if (property.hasOwnProperty(id)) {
+            uri = property[id].uri;
+            if (defined(uri) && !isDataUri(uri)) {
+                ++count;
+            }
+        }
+    }
+    return count;
+}
+
+function getNumberOfExternalRequests(gltf) {
+    var count = 0;
+
+    count += getNumberOfExternalProperties(gltf.buffers);
+    count += getNumberOfExternalProperties(gltf.images);
+    count += getNumberOfExternalProperties(gltf.shaders);
+
+    return count;
+}
+
+function getNumberOfProperties(object) {
+    return Object.keys(object).length;
+}
+
+function getDrawCallStatistics(gltf) {
+    var numberOfDrawCalls = 0;
+    var numberOfRenderedPrimitives = 0;
+
+    var primitives = PrimitiveHelpers.getAllPrimitives(gltf);
+    var primitivesLength = primitives.length;
+
+    numberOfDrawCalls += primitivesLength;
+
+    for (var m = 0; m < primitivesLength; ++m) {
+        var primitive = primitives[m];
+        var indices = primitive.indices;
+        var indicesCount = gltf.accessors[indices].count;
+
+        switch(primitive.primitive) {
+            case 0: // POINTS
+                numberOfRenderedPrimitives += indicesCount;
+                break;
+            case 1: // LINES
+                numberOfRenderedPrimitives += (indicesCount / 2);
+                break;
+            case 2: // LINE_LOOP
+                numberOfRenderedPrimitives += indicesCount;
+                break;
+            case 3: // LINE_STRIP
+                numberOfRenderedPrimitives += Math.max(indicesCount - 1, 0);
+                break;
+            case 4: // TRIANGLES
+                numberOfRenderedPrimitives += (indicesCount / 3);
+                break;
+            case 5: // TRIANGLE_STRIP
+            case 6: // TRIANGLE_FAN
+                numberOfRenderedPrimitives += Math.max(indicesCount - 2, 0);
+                break;
+        }
+    }
+
+    return {
+        numberOfDrawCalls : numberOfDrawCalls,
+        numberOfRenderedPrimitives : numberOfRenderedPrimitives
+    };
+}
+
+/**
+ * @typedef {Object} gltfStatistic
```

Comment:
Just to clarify, should I change this to be like a class? This would then look like:
```
function ComputeStatistics(...) {
    // properties here
}

ComputeStatistics.getStatistics() {
    // Code
    return ComputeStatistics(...);
}

ComputeStatistics.printStatistics() {
    // Code
}
```